src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs

```diff
@@ -809,9 +809,9 @@ public virtual int Read(Span<byte> buffer)
 
         public virtual int ReadByte()
         {
-            var oneByteArray = new byte[1];
-            int r = Read(oneByteArray, 0, 1);
-            return r == 0 ? -1 : oneByteArray[0];
+            Span<byte> oneByteSpan = stackalloc byte[1];
+            int r = Read(oneByteSpan);
```

Comment:
This needs to be reverted. It's changing the order of the base delegation and will result in stack overflows.